swim/node.go

```diff
@@ -218,7 +218,7 @@ func (n *Node) Incarnation() int64 {
 
 func (n *Node) emit(event interface{}) {
 	for _, listener := range n.listeners {
-		go listener.HandleEvent(event)
```

Comment:
I think we should have a clear definition of this contract.  Generally, handling events async seems  safer—it allows the event handler implementation to act freely without concern for slowing down other mission-critical paths.  Of course, that does require careful reasoning about how tightly two systems in question are coupled.  If they do need to be tightly coupled to the point of making  operations in the two subsystems atomic, then I think event handling might not be the right pattern.  If synchronous event handling is the right pattern, that contract had better be extremely clearly documented...